core/jazz_cloud-logs-streamer/test/test.js

```diff
@@ -14,14 +14,180 @@
+                expect(utils().isNumeric(25.6)).to.equal(true);
+            });
+            //negative test for isNumeric Function
+            it('should indicate a false numeric value', function () {
+                expect(utils().isNumeric("hello")).to.equal(false);
+            });
 
-        //Test cases to be added here.
-        assert(true);
-        done();
+        });
+        describe('hash', function () {
+            /* Given a string and a encoding technique, hash() returns an encoded output of given string
+            @params {String} any string value to be encoded
+            @params {String} any of the encoding techniques compatible with sha256 (hex, binary, base64)
+            @returns {String} encoded string
+            */
+
+            //encodes string = 'surya' in sha256 using hex encoding
+            it('should encode surya using hex', function () {
+                expect(utils().hash('surya', 'hex')).to.equal('ea1b9d779a37fa378d87c40dd6a56fcd491a7c9bef3a1f6e40228031bf00ac68');
+            });
+
+            //encodes string = 'surya' in sha256 using base64 encoding
+            it('should encode surya using base-64', function () {
+                expect(utils().hash('surya', 'base64')).to.equal('6hudd5o3+jeNh8QN1qVvzUkafJvvOh9uQCKAMb8ArGg=');
+            });
+
+            //Having trouble with special "\" in binary
+            //encodes string = 'surya' in sha256 using binary encoding
+            /*it('should encode surya using binary', function () {
+                var expected = 'ê' + '\\' + 'u001b7ú7Ä' + '\\' + 'rÖ¥oÍI' + '\\' + 'u001a|:' + '\\' + 'u001fn@"1¿' + '\\' + 'u0000¬h';
+                expect(utils().hash('surya', 'binary')).to.equal(expected);
+            });*/
+
+            //negative test condition to check for any invalid encoding technique
+            it('should return buffer for any encoding other than base64, hex, or binary', function () {
+                expect(utils().hash('surya', 'hex')).toString().includes("<Buffer");
+            });
+        });
+        describe('isValidJson', function () {
+            /* Given input, determine whether it is valid json or not
+            @params {String} input string
+            @returns {Boolean} true if given input string is a valid json file , false if given input string is not a valid json file
+            */
+
+            //positive test for isValidJson
+            it('should indicate a valid JSON', function () {
+                var json = '{ "Version": "2012-10-17","Statement": [{"Sid": "","Effect": "Allow","Principal": {"Service": "cloudtrail.amazonaws.com"},"Action": "sts:AssumeRole"}]}';
+                expect(utils().isValidJson(json)).to.equal(true);
+            });
+            it('should indicate a invalid JSON', function () {
+                var json = '{ "Version" "2012-10-17","Statement": [{"Sid": "","Effect": "Allow","Principal": {"Service": "cloudtrail.amazonaws.com"},"Action": "sts:AssumeRole"}]}';
```

Comment:
It isn't pretty obvious what makes this a negative test case. Try making test criteria simple where possible. At the same time keep looking for the corner cases where the test setup and/or criteria might be complex.

Reply:
ok surya will do